docs/designs/restore-and-gc.md

```diff
@@ -20,20 +20,40 @@ github:
   userCache: https:///some-blob-service/github-user-cache.json
 ```
 
-The restored files will be stored at `%DOCFX_APPDATA_PATH%/downloads/{url-short-path}/{version}+{etag}` like:
+The restored files will be stored at `%DOCFX_APPDATA_PATH%/downloads/{url-short-path}/{number}` like:
 
 ```text
 `%DOCFX_APPDATA_PATH%`
     | - downloads
         | - raw.gith..tent.com+docascode+docfx-te..ndencies+62b0448+extend1.yml+dc363b0e
-            | - 18b265bc4e9a6b4fdff8b5210caf9e3541f4cb2d+%22f02fde5b4f3817266458170ba048c7fc2de46287%22
-            | - 3111eb522b092c9cef5cd0ccd86892cf847592d6+%2299a5dbb8d73da1ec452d74c0ab45b11ebe8615a0%22
+            | - 1
+            | - 2
 
 ```
 
+And there is another index file `index.txt` under `%DOCFX_APPDATA_PATH%/downloads/{url-short-path}/` tracking the detail info of each `{number} file`.
+
+```text
+`%DOCFX_APPDATA_PATH%`
+    | - downloads
+        | - raw.gith..tent.com+docascode+docfx-te..ndencies+62b0448+extend1.yml+dc363b0e
+            | - 1
+            | - 2
+            | - index.txt
+
+```
+
+`index.txt`:
+```text
+1 {version} {etag} {date} {in-use}
+2 {version} {etag} {date} {in-use}
+```
+
 - `{url-short-path}` is calculated from `file` url
 - `{version}` is sha1 hash of file content, 
 - `{etag}` is timestamp or something else supported by the service which stores the `file`.
+- `{date}` is the download date, it will be set or overwritten during restore.
+- `{in-use}` is tracking the file is in use or not, if it is, it's can't be overwritten, otherwise, it can be overwritten for reusing during restore.
```

Comment:
what is the type of `in-use`?, it is used for reference counting, so at least it is an integer. To avoid corruption caused by process crash, maybe it could be a list of process ids.

Reply:
it's a string, "restoring" or "building". We already have ProcessUtility to tracking process info.